context/src/main/java/io/grpc/Context.java

```diff
@@ -99,6 +99,12 @@
   private static final PersistentHashArrayMappedTrie<Key<?>, Object> EMPTY_ENTRIES =
       new PersistentHashArrayMappedTrie<Key<?>, Object>();
 
+  // Long chains of contexts are suspicious and usually indicate a misuse of Context.
+  // The threshold and interval are arbitrarily chosen.
+  // VisibleForTesting
+  static final int CONTEXT_DEPTH_WARN_THRESH = 200;
```

Comment:
@adriancole, I'm okay with bumping this higher. If you're concerned about 200, do you have a suggestion on where to start, like 1000 or 5000?

The main concern at the moment is infinite mutation, because finding those in real life applications can be hard.